From 1fab8a81aa835f3395b46e8e6ff21d082fe1a048 Mon Sep 17 00:00:00 2001
From: pbuonado <pbuonado@meraki.com>
Date: Mon, 2 Jul 2018 16:48:28 -0700
Subject: [PATCH 05/26] 0993

---
 networking/ping.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/networking/ping.c b/networking/ping.c
index 8f85d3e..0cd1dee 100644
--- a/networking/ping.c
+++ b/networking/ping.c
@@ -732,6 +732,7 @@ static void ping4(len_and_sockaddr *lsa)
 	}
 
 	signal(SIGINT, print_stats_and_exit);
+	fflush(stdout);
 
 	/* start the ping's going ... */
  send_ping:
-- 
2.17.0

