commit 96b53e2bc42023bc858208be1066f76b4f1bac89
Author: Stephen Segal <ssegal@meraki.com>
Date:   Thu Jun 2 15:10:16 2016 -0700

    431-increase-max-password-length.patch

diff --git a/pppd/plugins/radius/radiusclient.h b/pppd/plugins/radius/radiusclient.h
index 51b959a..0a577b3 100644
--- a/pppd/plugins/radius/radiusclient.h
+++ b/pppd/plugins/radius/radiusclient.h
@@ -29,7 +29,7 @@ typedef int          INT4;
 #endif
 
 #define AUTH_VECTOR_LEN		16
-#define AUTH_PASS_LEN		(3 * 16) /* multiple of 16 */
+#define AUTH_PASS_LEN		(8 * 16) /* multiple of 16 */
 #define AUTH_ID_LEN		64
 #define AUTH_STRING_LEN		128	 /* maximum of 253 */
 
