--- a/configure	2012-07-31 09:17:26.000000000 -0700
+++ b/configure	2012-09-10 11:50:29.777224433 -0700
@@ -14144,13 +14144,12 @@
 
 
 # In case INADDR_NONE is not defined (like on Solaris)
-have_inaddr_none="no"
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for INADDR_NONE" >&5
 $as_echo_n "checking for INADDR_NONE... " >&6; }
 if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
+echo $? "cannot run test program while cross compiling
 See \`config.log' for more details" "$LINENO" 5; }
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -14521,7 +14520,7 @@
 if test "$cross_compiling" = yes; then :
   { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
+echo $? "cannot run test program while cross compiling
 See \`config.log' for more details" "$LINENO" 5; }
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
