diff --git a/dmiopt.c b/dmiopt.c
index 46e5cda..a719636 100644
--- a/dmiopt.c
+++ b/dmiopt.c
@@ -155,6 +155,7 @@ static const struct string_keyword opt_string_keyword[] = {
 	{ "system-version", 1, 0x06 },
 	{ "system-serial-number", 1, 0x07 },
 	{ "system-uuid", 1, 0x08 },             /* dmi_system_uuid() */
+	{ "system-family", 1, 0x1a },
 	{ "baseboard-manufacturer", 2, 0x04 },
 	{ "baseboard-product-name", 2, 0x05 },
 	{ "baseboard-version", 2, 0x06 },
-- 
1.7.4.2.g597a6

