--- a/configure.ac	2018-01-11 14:59:59.000000000 -0800
+++ b/configure.ac	2018-01-11 15:03:25.567933341 -0800
@@ -649,8 +649,8 @@
 	VGCONF_ARCH_PRI="arm"
 	VGCONF_PLATFORM_PRI_CAPS="ARM_LINUX"
 	VGCONF_PLATFORM_SEC_CAPS=""
-	valt_load_address_pri_norml="0x58000000"
-	valt_load_address_pri_inner="0x38000000"
+	valt_load_address_pri_norml="0x38000000"
+	valt_load_address_pri_inner="0x30000000"
         valt_load_address_sec_norml="0xUNSET"
         valt_load_address_sec_inner="0xUNSET"
 	AC_MSG_RESULT([ok (${host_cpu}-${host_os})])
