--- a/src/store_rewrite.c
+++ b/src/store_rewrite.c
@@ -134,7 +134,7 @@
 	fqdn = dash_str;
     xstrncpy(claddr, inet_ntoa(r->client_addr), 20);
     xstrncpy(myaddr, inet_ntoa(http->request->my_addr), 20);
-    snprintf(buf, 8191, "%s %s/%s %s %s %s myip=%s myport=%d",
+    snprintf(buf, 8191, "%s %s/%s %s %s %s myip=%s myport=%d sport=%d",
 	r->orig_url,
 	claddr,
 	fqdn,
@@ -142,7 +142,8 @@
 	r->method_s,
 	urlgroup ? urlgroup : "-",
 	myaddr,
-	http->request->my_port);
+	http->request->my_port,
+	http->request->client_port);
     debug(61, 6) ("storeurlStart: sending '%s' to the helper\n", buf);
     strcat(buf, "\n");
     helperSubmit(storeurlors, buf, storeurlHandleReply, r);
--- a/src/redirect.c	2008-05-04 16:23:13.000000000 -0700
+++ b/src/redirect.c	2013-05-17 14:31:48.222471482 -0700
@@ -134,7 +134,7 @@
 	fqdn = dash_str;
     xstrncpy(claddr, inet_ntoa(r->client_addr), 20);
     xstrncpy(myaddr, inet_ntoa(http->request->my_addr), 20);
-    snprintf(buf, 8191, "%s %s/%s %s %s %s myip=%s myport=%d",
+    snprintf(buf, 8191, "%s %s/%s %s %s %s myip=%s myport=%d sport=%d",
 	r->orig_url,
 	claddr,
 	fqdn,
@@ -142,7 +142,8 @@
 	r->method_s,
 	urlgroup ? urlgroup : "-",
 	myaddr,
-	http->request->my_port);
+	http->request->my_port,
+	http->request->client_port);
     debug(61, 6) ("redirectStart: sending '%s' to the helper\n", buf);
     strcat(buf, "\n");
     helperSubmit(redirectors, buf, redirectHandleReply, r);
