--- orig/common/discover.c	2011-10-20 14:24:42.000000000 -0700
+++ new/common/discover.c	2011-10-20 17:41:10.000000000 -0700
@@ -127,6 +127,7 @@
 	struct interface_info *tmp, *ip;
 	struct interface_info *last, *next;
-	char buf [2048];
+	char buf [30 * sizeof(struct ifreq)];
+	int reqlen;
 	struct ifconf ic;
 	struct ifreq ifr;
 	int i;
@@ -170,6 +171,7 @@
 #endif
 
       gifconf_again:
+	reqlen = ic.ifc_len;
 	i = ioctl(sock, SIOCGIFCONF, &ic);
 
 	if (i < 0)
@@ -201,6 +203,18 @@
 		ic.ifc_len = sizeof buf;
 		goto gifconf_again;
 #endif
+	} else if (ic.ifc_len > 0 && ic.ifc_len == reqlen) {
+		if (ic.ifc_ifcu.ifcu_buf != buf) {
+			dfree (ic.ifc_ifcu.ifcu_buf, MDL);
+			ic.ifc_ifcu.ifcu_buf = 0;
+		}
+		if (ic.ifc_len >= (128 * 1024))
+			log_fatal ("SIOCGIFCONF wants more than 128kbytes.");
+		ic.ifc_len *= 2;
+		ic.ifc_ifcu.ifcu_buf = dmalloc ((size_t)ic.ifc_len, MDL);
+		if (!ic.ifc_ifcu.ifcu_buf)
+			log_fatal ("Can't allocate SIOCGIFCONF buffer.");
+		goto gifconf_again;
 	}
 
 		
