--- origopenl2tp/plugins/ppp_unix.c	2012-02-16 17:17:47.000000000 -0800
+++ newopenl2tp/plugins/ppp_unix.c	2012-02-22 17:22:44.000000000 -0800
@@ -894,11 +894,9 @@
 
 		argv[arg++] = "plugin";
 		argv[arg++] = "pppol2tp.so";
-#ifdef L2TP_FEATURE_RPC_MANAGEMENT
 		/* This plugin requires pppol2tp.so to be loaded first */
 		argv[arg++] = "plugin";
 		argv[arg++] = "openl2tp.so";
-#endif
 
 		if (!ppp_unix_use_dummy_pppd()) {
 			int flags;
