diff --git a/http.c b/http.c
index dd8c9cf..2d7187c 100644
--- a/http.c
+++ b/http.c
@@ -2415,6 +2415,12 @@ evhttp_connection_get_addr(struct evhttp_connection *evcon)
	return (struct sockaddr *)evcon->conn_address;
 }

+size_t
+evhttp_connection_get_buffered_output_len(struct evhttp_connection *evcon)
+{
+	return evbuffer_get_length(bufferevent_get_output(evcon->bufev));
+}
+
 int
 evhttp_connection_connect_(struct evhttp_connection *evcon)
 {
diff --git a/include/event2/http.h b/include/event2/http.h
index 4284d5f..76c172f 100644
--- a/include/event2/http.h
+++ b/include/event2/http.h
@@ -593,6 +593,9 @@ EVENT2_EXPORT_SYMBOL
 void evhttp_request_set_error_cb(struct evhttp_request *,
     void (*)(enum evhttp_request_error, void *));

+/** Get the length of the buffered output associated with this connection. */
+size_t evhttp_connection_get_buffered_output_len(struct evhttp_connection *evcon);
+
 /**
  * Set a callback to be called on request completion of evhttp_send_* function.
  *
--
