--- a/src/parser.c
+++ b/src/parser.c
@@ -2409,7 +2409,7 @@ static void ParseRateFilter(SnortConfig *sc, SnortPolicy *p, char *args)
 
     if (RateFilter_Create(sc, sc->rate_filter_config,  &thdx))
     {
-        ParseError(ERR_CREATE);
+        ParseWarning(ERR_CREATE);
     }
 
     mSplitFree(&toks, num_toks);
@@ -9809,15 +9809,15 @@ static void ParseThreshFilter(
     {
         if (thdx.sig_id == 0)
         {
-            ParseError(ERR_CREATE_EX, "only one per gen_id != 0");
+            ParseWarning(ERR_CREATE_EX, "only one per gen_id != 0");
         }
         else if (thdx.gen_id == 0)
         {
-            ParseError(ERR_CREATE_EX, "gen_id = 0 requires sig_id = 0");
+            ParseWarning(ERR_CREATE_EX, "gen_id = 0 requires sig_id = 0");
         }
         else
         {
-            ParseError(ERR_CREATE_EX, "gen_id, sig_id must be unique");
+            ParseWarning(ERR_CREATE_EX, "gen_id, sig_id must be unique");
         }
     }
 
@@ -9950,7 +9950,7 @@ static void ParseSuppress(SnortConfig *sc, SnortPolicy *p, char *args)
 
     if (sfthreshold_create(sc, sc->threshold_config, &thdx))
     {
-        ParseError(ERR_CREATE);
+        ParseWarning(ERR_CREATE);
     }
 
     mSplitFree(&toks, num_toks);
