diff --git a/svr-authpubkey.c b/svr-authpubkey.c
index c3bac82..484c546 100644
--- a/svr-authpubkey.c
+++ b/svr-authpubkey.c
@@ -220,15 +220,6 @@ static int checkpubkey(char* algo, unsigned int algolen,
 		goto out;
 	}
 
-	/* we don't need to check pw and pw_dir for validity, since
-	 * its been done in checkpubkeyperms. */
-	len = strlen(ses.authstate.pw_dir);
-	/* allocate max required pathname storage,
-	 * = path + "/.ssh/authorized_keys" + '\0' = pathlen + 22 */
-	filename = m_malloc(len + 22);
-	snprintf(filename, len + 22, "%s/.ssh/authorized_keys", 
-				ses.authstate.pw_dir);
-
 	/* open the file as the authenticating user. */
 	origuid = getuid();
 	origgid = getgid();
@@ -237,7 +228,7 @@ static int checkpubkey(char* algo, unsigned int algolen,
 		dropbear_exit("Failed to set euid");
 	}
 
-	authfile = fopen(filename, "r");
+	authfile = fopen("/etc/dropbear/authorized_keys", "r");
 
 	if ((seteuid(origuid)) < 0 ||
 		(setegid(origgid)) < 0) {
@@ -348,7 +339,7 @@ static int checkpubkey(char* algo, unsigned int algolen,
 		ret = cmp_base64_key(keyblob, keybloblen, (const unsigned char *) algo, algolen, line, NULL);
 
 		if (ret == DROPBEAR_SUCCESS && options_buf) {
-			ret = svr_add_pubkey_options(options_buf, line_num, filename);
+			ret = svr_add_pubkey_options(options_buf, line_num, "/etc/dropbear/authorized_keys");
 		}
 
 		if (ret == DROPBEAR_SUCCESS) {
@@ -366,7 +357,6 @@ out:
 	if (line) {
 		buf_free(line);
 	}
-	m_free(filename);
 	if (options_buf) {
 		buf_free(options_buf);
 	}
@@ -381,8 +371,6 @@ out:
  * ~/.ssh/authorized_keys are all owned by either root or the user, and are
  * g-w, o-w */
 static int checkpubkeyperms() {
-
-	char* filename = NULL; 
 	int ret = DROPBEAR_FAILURE;
 	unsigned int len;
 
@@ -396,25 +384,7 @@ static int checkpubkeyperms() {
 		goto out;
 	}
 
-	/* allocate max required pathname storage,
-	 * = path + "/.ssh/authorized_keys" + '\0' = pathlen + 22 */
-	filename = m_malloc(len + 22);
-	strncpy(filename, ses.authstate.pw_dir, len+1);
-
-	/* check ~ */
-	if (checkfileperm(filename) != DROPBEAR_SUCCESS) {
-		goto out;
-	}
-
-	/* check ~/.ssh */
-	strncat(filename, "/.ssh", 5); /* strlen("/.ssh") == 5 */
-	if (checkfileperm(filename) != DROPBEAR_SUCCESS) {
-		goto out;
-	}
-
-	/* now check ~/.ssh/authorized_keys */
-	strncat(filename, "/authorized_keys", 16);
-	if (checkfileperm(filename) != DROPBEAR_SUCCESS) {
+	if (checkfileperm("/etc/dropbear/authorized_keys") != DROPBEAR_SUCCESS) {
 		goto out;
 	}
 
@@ -422,8 +392,6 @@ static int checkpubkeyperms() {
 	ret = DROPBEAR_SUCCESS;
 	
 out:
-	m_free(filename);
-
 	TRACE(("leave checkpubkeyperms"))
 	return ret;
 }
