Only in nmap-5.00.orig/: .#configure.ac
diff -ru nmap-5.00.orig/configure.ac nmap-5.00/configure.ac
--- nmap-5.00.orig/configure.ac	2009-06-24 14:35:54.000000000 -0700
+++ nmap-5.00/configure.ac	2009-08-19 14:06:50.000000000 -0700
@@ -368,26 +368,7 @@
   fi
 
   # link with -lpcap for the purposes of this test
-  LIBS_OLD="$LIBS"
   LIBS="$LIBS -lpcap"
-  AC_MSG_CHECKING(if libpcap version is recent enough)
-  AC_TRY_RUN([
-#include <stdio.h>
-extern char pcap_version[];
-int main() {
-  int major, minor1, minor2;
-  sscanf(pcap_version,"%d.%d.%d", &major, &minor1, &minor2);
-  if (major > 0) 
-    exit(0);
-  if (minor1 < 9)
-    exit(1);
-  if (minor2 < 4)
-    exit(1);
-  exit(0);
-}],
-[AC_MSG_RESULT(yes); have_libpcap=yes],
-[AC_MSG_RESULT(no); have_libpcap=no])
-LIBS="$LIBS_OLD"
 fi
 
 LIBPCAP_LIBS="-lpcap"
Only in nmap-5.00: configure.ac~
