diff --git a/options.h b/options.h
index 721e4e5..a3ab250 100644
--- a/options.h
+++ b/options.h
@@ -313,7 +313,7 @@ be overridden at runtime with -I. 0 disables idle timeouts */
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
-#define DEFAULT_PATH "/usr/bin:/bin"
+#define DEFAULT_PATH "/bin:/sbin:/usr/bin:/usr/sbin"
 
 /* Some other defines (that mostly should be left alone) are defined
  * in sysoptions.h */
