--- a/Samba/source/build/smb_build/makefile.pm	2011-07-26 07:35:19.331696720 -0700
+++ b/Samba/source/build/smb_build/makefile.pm	2011-07-26 07:36:24.391696740 -0700
@@ -136,6 +136,7 @@
 HOSTCC=$self->{config}->{HOSTCC}
 HOSTCC_CFLAGS=-D_SAMBA_HOSTCC_ $self->{config}->{CFLAGS} \$(CPPFLAGS)
 
+EXTRA_LINK_FLAGS ?=
 INSTALL_LINK_FLAGS=$extra_link_flags
 
 LD=$self->{config}->{LD} 
@@ -286,7 +287,7 @@
 $ctx->{SHAREDDIR}/$ctx->{LIBRARY_REALNAME}: \$($ctx->{TYPE}_$ctx->{NAME}_DEPEND_LIST) \$($ctx->{TYPE}_$ctx->{NAME}_FULL_OBJ_LIST) $init_obj
 	\@echo Linking \$\@
 	\@mkdir -p $ctx->{SHAREDDIR}
-	\@\$(SHLD) \$(SHLD_FLAGS) -o \$\@ \$(INSTALL_LINK_FLAGS) \\
+	\@\$(SHLD) \$(SHLD_FLAGS) -o \$\@ \$(INSTALL_LINK_FLAGS) \$(EXTRA_LINK_FLAGS) \\
 		\$($ctx->{TYPE}_$ctx->{NAME}\_FULL_OBJ_LIST) \\
 		\$($ctx->{TYPE}_$ctx->{NAME}_LINK_FLAGS) $extraflags \\
 		 $soarg \\
@@ -376,7 +377,7 @@
 $self->output(<< "__EOD__"
 $installdir/$ctx->{BINARY}: \$($ctx->{TYPE}_$ctx->{NAME}_DEPEND_LIST) \$($ctx->{TYPE}_$ctx->{NAME}_FULL_OBJ_LIST)
 	\@echo Linking \$\@
-	\@\$(LD) \$(LDFLAGS) -o \$\@ \$(INSTALL_LINK_FLAGS) \\
+	\@\$(CC) \$(LDFLAGS) -o \$\@ \$(INSTALL_LINK_FLAGS) \$(EXTRA_LINK_FLAGS) \\
 		\$\($ctx->{TYPE}_$ctx->{NAME}_LINK_FLAGS) 
 
 __EOD__
